src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,277 @@
+        }
+
+        Node<T> elementToAdd = new Node<>(value);
+
+        if (size == 0) {
+            head = elementToAdd;
+            tail = elementToAdd;
+        } else if (size - 1 == index) {
+            Node<T> elPrevTail = tail.prev;
+            elPrevTail.next = elementToAdd;
+            elementToAdd.prev = elPrevTail;
+
+            tail.prev = elementToAdd;
+            elementToAdd.next = tail;
+        } else if (size == index) {
+            tail.next = elementToAdd;
+            elementToAdd.prev = tail;
+            tail = elementToAdd;
+        } else if (index == 0 && size != 0) {
+            Node<T> currentElement = head;
+            head = elementToAdd;
+            elementToAdd.next = currentElement;
+            currentElement.prev = elementToAdd;
+        } else {
+            int countOfIndex = 0;
+            Node<T> element = head;
+
+            while (countOfIndex != index) {
+                element = element.next;
+                countOfIndex++;
+            }
+
+            Node<T> elPrevElement = element.prev;
+            elPrevElement.next = elementToAdd;
+            elementToAdd.prev = elPrevElement;
+
+            elementToAdd.next = element;
+            element.prev = elementToAdd;
+        }
+
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (!list.isEmpty()) {
+            for (T element : list) {
+                Node<T> newElement = new Node<>(element);
+                tail.next = newElement;
+                newElement.prev = tail;
+                tail = newElement;
+                size++;
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " does not exist");
+        }
+
+        if (size == 1) {
+            return head.item;
+        } else {
+            int countOfIndex = 0;
+            Node<T> elementToReturn = head;
+
+            while (countOfIndex != index) {
+                elementToReturn = elementToReturn.next;
+                countOfIndex++;
+            }
+
+            return elementToReturn.item;
+        }
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " does not exist");
+        }
+
+        Node<T> elementToSet = new Node<>(value);
+        Node<T> elementToReturn = null;
+
+        if (size == 1) {
+            elementToReturn = head;
+            head = elementToSet;
+            tail = elementToSet;
+        } else if (index == size - 1) {
+            elementToReturn = tail;
+            Node<T> elPrevTail = tail.prev;
+            elPrevTail.next = elementToSet;
+            elementToSet.prev = elPrevTail;
+            tail = elementToSet;
```

Comment:
In the `set` method, when replacing the tail, ensure that the `next` pointer of the new tail is set to `null`.